src/libcollections/string.rs

```diff
@@ -1015,8 +1015,7 @@ impl String {
     }
 
     /// Create a draining iterator that removes the specified range in the string
-    /// and yields the removed chars from start to end. The element range is
-    /// removed even if the iterator is not consumed until the end.
```

Comment:
Not sure I understand you. Following does not panic:

    let mut v = vec![1, 2, 3];
    assert_eq!(vec![3], v.drain(2..).collect::<Vec<_>>());
    assert_eq!(vec![1, 2], v);
    v.drain(..);
    assert!(v.is_empty());